Filter layer quantifiers and add test

DragaDoncila opened the pull request:
As per #721 this PR updates `parse_manifest` to filter out layer type quantifiers in the `writer_save_layers` field. I've also added a test for this explicit bug.

Preview page demonstrating filtering:
https://preview.napari-hub.org/chanzuckerberg/napari-demo/19

DragaDoncila commented:
Ugh don't think we can get rid of that test failure. It's failing because the preview metadata doesn't match the hub metadata, but this is exactly what this PR is fixing. So this test will only pass after we merge this PR. Probably a sign it's not a great test and we should rewrite it. I think that's out of scope for this PR

DragaDoncila commented:
This does make me wonder if the metadata on the backend will be updated as a result of this change though. Afaik we would only invoke updates if the plugin itself changes? Do we have to delete all the manifests after merging this PR *JUST* to trigger a rebuild of the plugin metadata in `index.json`? @potating-potato @richaagarwal?

manasaV3 commented on backend/utils/utils.py:
```diff
@@ -13,6 +14,17 @@
+    'tracks',
+    'vectors',
+]
+VALID_LAYER_REGEX = rf"({'|'.join([layer_type for layer_type in VALID_LAYERS])}).*"
```
`rf` string seems really cool. :)

DragaDoncila commented on backend/utils/_tests/test_utils.py:
```diff
@@ -0,0 +1,22 @@
+            ]
+    }}
+    parsed_attributes = parse_manifest(manifest)
+    assert sorted(parsed_attributes['writer_save_layers']) == sorted(['image', 'labels', 'points', 'tracks'])
```
Nope, you're right. Fixed that.

DragaDoncila commented on backend/utils/_tests/test_utils.py:
```diff
@@ -0,0 +1,22 @@
+                        'faketype',
+                        'points+',
+                        'tracks*'
+                    ]
```
I think we *should* test against all valid layer types, but I don't think this should be done here as this increases maintenance burden for us and will require updating as new layer types get released. I think testing of all valid layer types should occur in the same test for both `backend` and `frontend` and it should get tested against napari layer types directly imported from napari.

DragaDoncila commented:
Ok I've double checked the update mechanism and we shouldn't need to delete anything for `index.json` to be updated. We will double check on staging that this is the case - worst case scenario if we find `index.json` is not updated on staging, we can delete it at any time.

@manasaV3 @codemonkey800 I've addressed comments I believe so would appreciate a re-review here!

manasaV3 commented:
From my understanding of the process that caches the manifest data, it will automatically write for the latest versions of plugins where the manifest data is missing. For the older plugin versions, we would have to manually trigger it to ensure we keep historical data.

DragaDoncila commented:
I'm going to merge this now as @manasaV3 has approved and I'm pretty confident with the changes. I'm going to monitor `index.json` to make sure the changes are reflected and will ping back here when that's done.

DragaDoncila commented:
Have confirmed this works as expected for the `plugins/{plugin}` endpoint. `index.json` hasn't updated it. Will check back in but we may need a manual delete of that one.
